packages/dnb-eufemia/src/components/list-format/ListFormat.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useContext, useMemo } from 'react'
+import { LOCALE } from '../../shared/defaults'
+import {
+  convertJsxToString,
+  extendPropsWithContext,
+} from '../../shared/component-helper'
+import SharedContext, { InternalLocale } from '../../shared/Context'
+import { Li, Ol, Ul } from '../../elements'
+
+export type ListFormatProps = {
+  /**
+   * Formatting options for the value.
+   * See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat/ListFormat
+   */
+  format?: Intl.ListFormatOptions
+  /**
+   * Defines if the value should be displayed in list format or regular text format on one line.
+   * Default: `text`
+   */
+  variant?: 'ol' | 'ul' | 'text'
+  /**
+   * Defines the type of list styling used for list variants. Used together with variant `ol` and `ul`.
+   * Variant `ol`: `a`, `A`, `i`, `I` and `1`.
+   * Variant `ul`: `circle`, `disc` and `square`.
+   * Default: `undefined`
+   */
+  listType?:
+    | 'a'
+    | 'A'
+    | 'i'
+    | 'I'
+    | '1'
+    | 'circle'
+    | 'disc'
+    | 'square'
+    | undefined
+
+  /**
+   * The value to format as list.
+   * Default: null
+   */
+  value?: Array<React.ReactNode>
+
+  /**
+   * The children to format as list.
+   * Default: null
+   */
+  children?: React.ReactNode
+}
+
+function ListFormat(localProps: ListFormatProps) {
+  const { locale, ListFormat } = useContext(SharedContext)
+
+  // Extract additional props from global context
+  const allProps = extendPropsWithContext(localProps, {}, ListFormat)
+  const { value, format, variant = 'text', listType, children } = allProps
+
+  const list = useMemo(() => {
+    const isListVariant = variant !== 'text'
+    if (children) {
+      return isListVariant
+        ? React.Children.map(children, (child: React.ReactNode, index) => {
+            return <Li key={index}>{child}</Li>
+          })
+        : children
+    }
+    return isListVariant
+      ? value?.map((value, index) => (
+          <Li key={index}>
+            {React.isValidElement(value)
+              ? value
+              : convertJsxToString(value)}
+          </Li>
+        ))
+      : value
+  }, [value, children, variant])
+
+  const result = useMemo(() => {
+    if (variant === 'text') {
+      return listFormat(list, { locale, format })
+    }
+
+    const ListElement = variant.startsWith('ol') ? Ol : Ul
+
+    return <ListElement type={listType}>{list}</ListElement>
+  }, [format, list, locale, variant, listType])
+
+  return result
+}
+
+// Support for "ListFormat.format(list)" for non-React usage
+ListFormat.format = listFormat
+
+export function listFormat(
+  list: Array<React.ReactNode> | React.ReactNode,
```

Comment:
```suggestion
  list: Array<React.ReactNode>,
```

I think it always should be an array.

Reply:
I've tried to make the change now.
But not 100% sure  how I'll make it work, as I think we would have to support React.ReactNode, or else we'll have to do some more manipulation before passing children to this function

Reply:
Resolved it now, when recieving a single React.ReactNode, I jsut treat it as [React.ReactNode]